### DISCUSSION TITLE
How can i change the extension of a `*.verified.*` result file

### DISCUSSION THREAD

**Discussion by samtrion** [VerifyTests/Verify#1216] [Q&A]
In one of my projects, i try to generate `.csproj` files. I check these files with `_ = await Verifier.VerifyFile(Builder.FullName, Predefined.VerifySettings).UseParameters(targetFramework.Name);`

How can I change the result file `*.verified.csproj` to something with less problems?

`Predefined.VerifySettings` looks like ...
```csharp

internal static class Predefined
{
    private static VerifySettings? _verifySettings;
    public static VerifySettings VerifySettings
    {
        get
        {
            if (_verifySettings == null)
            {
                _verifySettings = new VerifySettings();
                _verifySettings.UseDirectory(GetSnapshotDirectory());
            }
            return _verifySettings;
        }
    }

    private static string GetSnapshotDirectory()
    {
        var directoryInfo = new DirectoryInfo(".");
        var assemblyName = typeof(Predefined).Assembly.GetName().Name;

        do
        {
            if (directoryInfo.Parent is null)
            {
                throw new DirectoryNotFoundException();
            }
            directoryInfo = directoryInfo.Parent;
        } while (!directoryInfo.Name.Equals(assemblyName, StringComparison.OrdinalIgnoreCase));

        directoryInfo = directoryInfo.CreateSubdirectory("_snapshots");

        return directoryInfo.FullName;
    }
}
```

**Comment by SimonCropp** [comment]
regarding your predefined. try this instead


    [ModuleInitializer]
    public static void Init()
    {
        Verifier.DerivePathInfo(
            (sourceFile, projectDirectory, method, type) =>
            {
                var snapshots = Path.Combine(projectDirectory, "_snapshots");
                Directory.CreateDirectory(snapshots);
                return new(snapshots, type.Name, method.Name);
            });
    }

docs https://github.com/VerifyTests/Verify/blob/main/docs/naming.md#derivepathinfo

then for the file name extension. currently you cant. i will add that functionality. should ship tomorrow

**Reply by samtrion** [reply]
@SimonCropp Thanks for the tip. Unfortunately I can only replace a part of the `Predefined` class, because I still use `HashParameters()`. If this could also be administered centrally, it would be great.

**Reply by SimonCropp** [reply]
i added static HashParameters in 25.0.0-beta.2   https://github.com/VerifyTests/Verify/blob/main/docs/parameterised.md#globally

**Comment by SimonCropp** [comment]
can you update to V 25.0.0-beta.1 and try this 
https://github.com/VerifyTests/Verify/blob/main/docs/verify-file.md#using-a-custom-extension

**Reply by samtrion** [reply]
Thanks, will try this later today.

**Reply by samtrion** [reply]
@SimonCropp Works great, thank you very much. When will it be officially available?

**Reply by SimonCropp** [reply]
> When will it be officially available

monday at the latest

**Reply by SimonCropp** [reply]
stable is on nuget
